### PR TITLE
NIFI-10153 Refactor DocGeneratorTest and remove binary test files

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-10153](https://issues.apache.org/jira/browse/NIFI-10153) Refactors the `DocGeneratorTest` in `nifi-documentation` and to eliminate the need for binary NAR files in the project repository. The refactored implementation mocks the `ExtensionManager` and returns component and bundle information to support HTML documentation generation.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java [hunk 1]
```diff
@@ -19,100 +19,87 @@
...
+        final BundleCoordinate bundleCoordinate = BundleCoordinate.UNKNOWN_COORDINATE;
+        final BundleDetails bundleDetails = new BundleDetails.Builder().workingDir(workingDirectory).coordinate(bundleCoordinate).build();
+        final Bundle bundle = new Bundle(bundleDetails, getClass().getClassLoader());
+        final ExtensionDefinition definition = new ExtensionDefinition(PROCESSOR_CLASS.getName(), bundle, Processor.class);
+        final Set<ExtensionDefinition> extensions = Collections.singleton(definition);
+        when(extensionManager.getExtensions(eq(Processor.class))).thenReturn(extensions);
+        doReturn(PROCESSOR_CLASS).when(extensionManager).getClass(eq(definition));
+
+        final Processor processor = new ProcessorWithLogger();
+        when(extensionManager.getTempComponent(eq(PROCESSOR_CLASS.getName()), eq(bundleCoordinate))).thenReturn(processor);
```
Comment (bejancsaba) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java:86]: As you don't really need the processor you could just simply do Mockito.mock(ProcessorWithLogger.class) at return.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java [hunk 2]
```diff
@@ -19,100 +19,87 @@
...
+
+        DocGenerator.generate(properties, extensionManager, extensionMapping);
+
+        final Collection<File> files = FileUtils.listFiles(workingDirectory, HTML_EXTENSIONS, RECURSIVE_ENABLED);
+        assertFalse(files.isEmpty());
+
+        final File file = files.iterator().next();
+        final byte[] bytes = Files.readAllBytes(file.toPath());
```
Comment (bejancsaba) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java:94]: Can you use Files.readString? It is basically the same you can save a few lines :)

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java (hunk 1 above)
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java:86]: Actually the documentation generation uses the instantiated component during HTML generation.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java (hunk 2 above)
Comment (exceptionfactory) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/test/java/org/apache/nifi/documentation/DocGeneratorTest.java:94]: It looks like [Files.readString](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Files.html#readString(java.nio.file.Path)) was introduced in Java 11, so this seems to be the best approach for now.

**Comment by exceptionfactory** [comment]
Thanks for the review @bejancsaba!

**Review by kevdoran — APPROVED** [review]
+1 from me. Thanks @exceptionfactory - if you can address the merge conflicts, I'll be happy to merge this

**Comment by exceptionfactory** [comment]
Thanks @kevdoran, rebased.

**Pull request closed without merging**